### ISSUE TITLE
Add codesections to «meta» reviewers

### ISSUE THREAD

**Issue by codesections** [Raku/problem-solving#241]
If appropriate to do so, please add me to the list of reviewers in the «meta» category.

Please let me know if I should provide any additional information (the `README` doesn't specify criteria for becoming a reviewer).

**Comment by AlexDaniel** [comment]
Sure, please submit a PR.

**Comment by AlexDaniel** [comment]
Ah, also, “fallback – if no other label fits” – this should also belong to you.

**Comment by AlexDaniel** [comment]
Also, I want to clarify something, and also make sure that you specifically understand what this process is about and how it works. Moreover, I like the shortness of the README but I wasn't direct enough when I was writing it, and as a result a lot of process details are implicitly derived from what is written indirectly. It'd be nice if you changed the README (but I'd strongly suggest to keep its length, because clearly most are not even reading what is already there). Just as an example of what I did badly, it says “If a reviewer does not respond in 14 days, they are removed from the list of reviewers on that PR.” – what this means in real life:
* A PR is merged in 14 days if nobody rejects it
* A PR is merged Later (after 14 days) or never if somebody does
* Or faster if *all* reviewers accept it, but nowadays that's unrealistic.

Now, the important bit. A reviewer here isn't attached to a label, instead a reviewer is supposed to review *all* pull requests to this repository. That's a very high standard, but it is meant to be so. You may find that surprising, but it seems to work in practice very well.

The list you're talking about serves a different purpose, or at least meant to serve a different purpose. Unfortunately, I never found a word that would describe it best. @ugexe used to say “subject-matter experts” (SMEs), but I simply said “developers responsible for a particular label”. Or, maybe incorrectly, “responsible devs” :) Point is, these are the people who are supposed to define a direction and drive the thought process and progress of a particular area. They cannot force a certain way of doing things, but they can provide their wisdom to show what should be done, given their knowledge and experience. That's why tickets are assigned to them, so that they can use their view to quickly decide on the direction, and later reviewers will decide what can and cannot be done.

Let's take this ticket as an example. The title should've been a bit different, as you didn't define X of the XY (as mentioned in the README). But that's OK. The issue here is “AlexDaniel's departure leaves some labels with no SMEs”. Now, I'm assigned to this ticket, and I can quickly tell you the solution that will likely work (given my understanding), and I can provide quick directions on how to achieve a useful (in my opinion) PR.

After you submit it, people in the [list of reviewers](https://github.com/Raku/problem-solving#reviewers) will approve or reject it. In 14 days it'll likely be merged.

It' is not that hard, and looking at it again with a fresh mind, it is actually an effective process that minimizes effort of all parties and yet creates a high standard for important changes while making sure that the hell does not break loose. But I really failed to communicate the way it should work, which is why I highly recommend changing the README to make it much clearer.

**Comment by AlexDaniel** [comment]
Also, for the future, when tweaking the README, you should know about [this ticket](https://github.com/Raku/problem-solving/issues/200) in particular. The ticket says “I'm not blaming @jnthn, I'm blaming the process”. I personally blame myself for not writing a clearer document that would show what SMEs were supposed to do. Read the ticket for some context. I expected SMEs to leave short and actionable comments like “do this, don't do that”, “explore this approach, if it doesn't work we'll try that”, etc. That is, I did not intend that SMEs would write large open-ended blogpost-like comments that leave everyone with burnout. At some point I did ask jnthn to write shorter responses, to which they said that it is easier for them to write longer ones… I guess you can see very well that I'm not a great communicator, and I wasn't able to explain how this process can work.

**Comment by codesections** [comment]
Thanks, that's all very helpful (especially the terminological note about the distinction between "reviewers" and "responsible devs"/"SMEs" – you're absolutely right that I was conflating the two).

One question based on what you said above/the `README` text:  You mentioned that "a PR is merged in 14 days if nobody rejects it".  Is that something that happens _automatically_?  Or is it just the case that, under the procedures in the `README`, anyone with commit access is allowed to merge the PR if no one rejects it for 14 days?

**Comment by AlexDaniel** [comment]
@codesections well, I think it's the role of devs responsible for the `meta` label to enforce the “rules” so to say. There is no automation for it yet, and we didn't have that many PRs here anyway to actually justify the creation of tools for that. I also don't think that anyone with commit access can actually merge anything based on the way it is configured now, only those with admin (“owner”) access.

**Comment by codesections** [comment]
Thanks, that's very helpful to understand :+1:

**Comment by AlexDaniel** [comment]
By the way, I think it is currently not defined when these 14 days start officially, although that mostly follows common sense. That said, ideally, the SME who asked to create the PR should indicate that the PR is what they wanted, and perhaps they should mark it as “ready for review”. Something to think about, maybe.

**Comment by AlexDaniel** [comment]
OK, I just learned about the CODEOWNERS file! Added it to the repo and ticket the box that makes reviews from code owners mandatory. This is how non-draft pull requests will appear:

![image](https://user-images.githubusercontent.com/5507503/97693323-aed63080-1aa9-11eb-9ba9-a61bbe9fc184.png)

Basically, it says “Waiting on code owner review from AlexDaniel”, and in the future you can use that to mark the beginning of the 14 days. For example, if unicode-related PR is submitted, then it will say “Waiting on code owner review from samcv”, besides also waiting for reviews from other reviewers. Does it make sense?

@codesections that file is currently not part of the PR so please once your PR is merged just fix the file directly. Also, please do not change the PR, as it invalidates all reviews and essentially resets the problem-solving timer.
